packages/viewer/src/components/scene-tree-table/scene-tree-table.tsx

```diff
@@ -0,0 +1,511 @@
+import {
+  Component,
+  Host,
+  h,
+  Prop,
+  State,
+  Element,
+  Method,
+} from '@stencil/core';
+import { Node } from '@vertexvis/scene-tree-protos/scenetree/protos/domain_pb';
+import { readDOM } from '../../lib/stencil';
+import { Binding } from '../scene-tree/lib/binding';
+import { SceneTreeController } from '../scene-tree/lib/controller';
+import { getSceneTreeViewportHeight } from '../scene-tree/lib/dom';
+import { ElementPool } from '../scene-tree/lib/element-pool';
+import { LoadedRow, Row } from '../scene-tree/lib/row';
+import {
+  generateInstanceFromTemplate,
+  InstancedTemplate,
+} from '../scene-tree/lib/templates';
+import { RowDataProvider } from '../scene-tree/scene-tree';
+import { getSceneTreeTableOffsetTop } from './lib/dom';
+
+interface StateMap {
+  columnElementPools?: WeakMap<
+    HTMLVertexSceneTreeTableColumnElement,
+    ElementPool
+  >;
+  headerInstances?: Array<InstancedTemplate<HTMLElement>>;
+
+  startIndex: number;
+  endIndex: number;
+  viewportRows: Row[];
+}
+
+@Component({
+  tag: 'vertex-scene-tree-table',
```

Comment:
How about renaming this to `vertex-scene-tree-table-layout`? The current name implies to me this can be used in place of `vertex-scene-tree`.